Volume mounts not working for Spark dependencies job

I am trying to pass a java trustStore file which contains the ES root ca cert to the Spark dependencies job to get the connection to ES work over TLS, but it looks like the volumes are not getting mounted and the job fails with the following error.

```
Caused by: org.elasticsearch.hadoop.EsHadoopIllegalStateException: Cannot initialize SSL - Expected to find keystore file at [file:/es-tls/trust.store] but was unable to. Make sure that it is available on the classpath, or if not, that 
you have specified a valid URI.
```

My dependencies config snippet in the Jaeger custom resource
 ```
dependencies:
      enabled: true
      javaOpts: -Djavax.net.ssl.trustStore=/es-tls/trust.store -Djavax.net.ssl.trustStorePassword='<pass>'
      resources:
        limits:
          memory: 4096Mi
        requests:
          memory: 2048Mi
      schedule: 16 05 * * *
      sparkMaster: ""
      volumeMounts:
      - mountPath: /es-tls
        name: es-tls
        readOnly: true
        subPath: ""
      volumes:
      - configMap:
          name: es-tls
        name: es-tls
```

This is what the job looks like when I describe the job and the volumemount is missing there
```
$ oc describe job jaeger-kafka-spark-dependencies-1607318160
Name:           jaeger-kafka-spark-dependencies-1607318160
Namespace:      tracing
Selector:       controller-uid=dcc93485-9db4-4251-954b-b842e98a0774
Labels:         app=jaeger
                app.kubernetes.io/component=spark-dependencies
                app.kubernetes.io/instance=jaeger-kafka
                app.kubernetes.io/managed-by=jaeger-operator
                app.kubernetes.io/name=jaeger-kafka-spark-dependencies
                app.kubernetes.io/part-of=jaeger
                controller-uid=dcc93485-9db4-4251-954b-b842e98a0774
                job-name=jaeger-kafka-spark-dependencies-1607318160
Annotations:    <none>
Controlled By:  CronJob/jaeger-kafka-spark-dependencies
Parallelism:    1
Completions:    <unset>
Start Time:     Mon, 07 Dec 2020 06:16:03 +0100Pods Statuses:  1 Running / 0 Succeeded / 3 Failed
Pod Template:
  Labels:           app=jaeger
                    app.kubernetes.io/component=spark-dependencies
                    app.kubernetes.io/instance=jaeger-kafka
                    app.kubernetes.io/managed-by=jaeger-operator
                    app.kubernetes.io/name=jaeger-kafka-spark-dependencies
                    app.kubernetes.io/part-of=jaeger
                    controller-uid=dcc93485-9db4-4251-954b-b842e98a0774
                    job-name=jaeger-kafka-spark-dependencies-1607318160
  Annotations:      linkerd.io/inject: disabled
                    prometheus.io/scrape: false
                    sidecar.istio.io/inject: false
  Service Account:  jaeger-kafka
  Containers:
   jaeger-kafka-spark-dependencies:
    Image:      jaegertracing/spark-dependencies
    Port:       <none>
    Host Port:  <none>
    Limits:
      memory:  4Gi
    Requests:
      memory:  2Gi
    Environment Variables from:
      jaeger-es-cred  Secret  Optional: false
    Environment:
      STORAGE:    elasticsearch
      JAVA_OPTS:  -Djavax.net.ssl.trustStore=/es-tls/trust.store -Djavax.net.ssl.trustStorePassword='<pass>'
      ES_NODES:   https://logging-es-http.logging.svc:9200
    Mounts:       <none>
  Volumes:        <none>
Events:
  Type    Reason            Age   From            Message
  ----    ------            ----  ----            -------
  Normal  SuccessfulCreate  73s   job-controller  Created pod: jaeger-kafka-spark-dependencies-1607318160-bvjh7
  Normal  SuccessfulCreate  65s   job-controller  Created pod: jaeger-kafka-spark-dependencies-1607318160-ln46g
  Normal  SuccessfulCreate  55s   job-controller  Created pod: jaeger-kafka-spark-dependencies-1607318160-djsjl
  Normal  SuccessfulCreate  34s   job-controller  Created pod: jaeger-kafka-spark-dependencies-1607318160-sz2x2
```
I followed this approach after referring to - 
https://github.com/jaegertracing/jaeger-operator/issues/294#issuecomment-616642204

--- a/pkg/cronjob/spark_dependencies.go
+++ b/pkg/cronjob/spark_dependencies.go
@@ -96,6 +96,7 @@ func CreateSparkDependencies(jaeger *v1.Jaeger) *batchv1beta1.CronJob {
 									Env:       util.RemoveEmptyVars(envVars),
 									EnvFrom:   envFromSource,
 									Resources: commonSpec.Resources,
+									VolumeMounts: jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.VolumeMounts,
 								},
 							},
 							RestartPolicy:      corev1.RestartPolicyNever,
@@ -103,6 +104,7 @@ func CreateSparkDependencies(jaeger *v1.Jaeger) *batchv1beta1.CronJob {
 							Tolerations:        commonSpec.Tolerations,
 							SecurityContext:    commonSpec.SecurityContext,
 							ServiceAccountName: account.JaegerServiceAccountFor(jaeger, account.DependenciesComponent),
+							Volumes:			jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.Volumes,
 						},
 						ObjectMeta: metav1.ObjectMeta{
 							Labels:      commonSpec.Labels,

--- a/pkg/cronjob/spark_dependencies.go
+++ b/pkg/cronjob/spark_dependencies.go
@@ -93,9 +93,9 @@ func CreateSparkDependencies(jaeger *v1.Jaeger) *batchv1beta1.CronJob {
 									Image: image,
 									Name:  name,
 									// let spark job use its default values
-									Env:       util.RemoveEmptyVars(envVars),
-									EnvFrom:   envFromSource,
-									Resources: commonSpec.Resources,
+									Env:          util.RemoveEmptyVars(envVars),
+									EnvFrom:      envFromSource,
+									Resources:    commonSpec.Resources,
 									VolumeMounts: jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.VolumeMounts,
 								},
 							},
@@ -104,7 +104,7 @@ func CreateSparkDependencies(jaeger *v1.Jaeger) *batchv1beta1.CronJob {
 							Tolerations:        commonSpec.Tolerations,
 							SecurityContext:    commonSpec.SecurityContext,
 							ServiceAccountName: account.JaegerServiceAccountFor(jaeger, account.DependenciesComponent),
-							Volumes:			jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.Volumes,
+							Volumes:            jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.Volumes,
 						},
 						ObjectMeta: metav1.ObjectMeta{
 							Labels:      commonSpec.Labels,

--- a/pkg/cronjob/spark_dependencies_test.go
+++ b/pkg/cronjob/spark_dependencies_test.go
@@ -224,3 +224,37 @@ func TestCustomSparkDependenciesImage(t *testing.T) {
 	assert.Empty(t, jaeger.Spec.Storage.Dependencies.Image)
 	assert.Equal(t, "org/custom-spark-dependencies-image", cjob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].Image)
 }
+
+func TestDependenciesVolumes(t *testing.T) {
+	testVolumeName := "testDependenciesVolume"
+	testConfigMapName := "dvConfigMap"
+	jaeger := v1.NewJaeger(types.NamespacedName{Name: "TestDependenciesVolumes"})
+	testVolume := corev1.Volume{
+		Name:         testVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap:             &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: testConfigMapName},
+			},
+		},
+	}
+	testVolumes := []corev1.Volume{testVolume}
+	jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.Volumes=testVolumes
+
+	testVolumeMountName := "testVolumeMount"
+	testMountPath := "/es-tls"
+	testVolumeMount := corev1.VolumeMount{
+		Name:             testVolumeMountName,
+		ReadOnly:         false,
+		MountPath:        testMountPath,
+	}
+	testVolumeMounts := []corev1.VolumeMount{testVolumeMount}
+	jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.VolumeMounts = testVolumeMounts
+
+	cjob := CreateSparkDependencies(jaeger)
+	assert.Equal(t, testVolumeMountName, cjob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].VolumeMounts[0].Name)
+	assert.False(t, cjob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].VolumeMounts[0].ReadOnly)
+	assert.Equal(t, testMountPath, cjob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].VolumeMounts[0].MountPath)
+
+	assert.Equal(t, testVolumeName, cjob.Spec.JobTemplate.Spec.Template.Spec.Volumes[0].Name)
+	assert.Equal(t, testConfigMapName, cjob.Spec.JobTemplate.Spec.Template.Spec.Volumes[0].ConfigMap.Name)
+}

--- a/pkg/cronjob/spark_dependencies_test.go
+++ b/pkg/cronjob/spark_dependencies_test.go
@@ -230,22 +230,22 @@ func TestDependenciesVolumes(t *testing.T) {
 	testConfigMapName := "dvConfigMap"
 	jaeger := v1.NewJaeger(types.NamespacedName{Name: "TestDependenciesVolumes"})
 	testVolume := corev1.Volume{
-		Name:         testVolumeName,
+		Name: testVolumeName,
 		VolumeSource: corev1.VolumeSource{
-			ConfigMap:             &corev1.ConfigMapVolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
 				LocalObjectReference: corev1.LocalObjectReference{Name: testConfigMapName},
 			},
 		},
 	}
 	testVolumes := []corev1.Volume{testVolume}
-	jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.Volumes=testVolumes
+	jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.Volumes = testVolumes
 
 	testVolumeMountName := "testVolumeMount"
 	testMountPath := "/es-tls"
 	testVolumeMount := corev1.VolumeMount{
-		Name:             testVolumeMountName,
-		ReadOnly:         false,
-		MountPath:        testMountPath,
+		Name:      testVolumeMountName,
+		ReadOnly:  false,
+		MountPath: testMountPath,
 	}
 	testVolumeMounts := []corev1.VolumeMount{testVolumeMount}
 	jaeger.Spec.Storage.Dependencies.JaegerCommonSpec.VolumeMounts = testVolumeMounts

